controllers/src/main/scala/cromwell/pipeline/controller/AkkaHttpClient.scala

```diff
@@ -0,0 +1,71 @@
+package cromwell.pipeline.controller
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model.Uri.Query
+import akka.http.scaladsl.model.headers.RawHeader
+import akka.http.scaladsl.model.{ HttpMethods, HttpRequest, HttpResponse, Uri }
+import akka.stream.ActorMaterializer
+import cromwell.pipeline.service.{ HttpClient, Response }
+
+import scala.concurrent.duration._
+import scala.concurrent.{ ExecutionContext, Future }
+
+class AkkaHttpClient(implicit actorSystem: ActorSystem, materializer: ActorMaterializer) extends HttpClient {
+
+  override def get(url: String, params: Map[String, String] = Map(), headers: Map[String, String] = Map())(
+    implicit ec: ExecutionContext
+  ): Future[Response] = {
+    val futureResponse: Future[HttpResponse] = Http().singleRequest(
+      HttpRequest(method = HttpMethods.GET, uri = Uri(url).withQuery(Query(params))).withHeaders(parseHeaders(headers))
+    )
+    responsify(futureResponse)
+  }
+
+  override def post(
+    url: String,
+    params: Map[String, String] = Map(),
+    headers: Map[String, String] = Map(),
+    payload: String
+  )(implicit ec: ExecutionContext): Future[Response] = {
+    val futureResponse: Future[HttpResponse] = Http().singleRequest(
+      HttpRequest(method = HttpMethods.POST, uri = Uri(url).withQuery(Query(params)))
+        .withEntity(payload)
+        .withHeaders(parseHeaders(headers))
+    )
+    responsify(futureResponse)
+  }
+
+  private def responsify(futureResponse: Future[HttpResponse])(implicit ec: ExecutionContext): Future[Response] =
+    futureResponse.flatMap { response =>
+      response.entity.toStrict(300.millis).map { body =>
```

Comment:
Avoid magic numbers.
You can make it a private class field or you can even take it as an implicit parameter for a whole class, but I'm not sure whether it would be overkill or not.
@ilyaonishenko how do you think?

Reply:
I would move it to the configuration file.

Reply:
Yep, I thought so too. I'm just not sure that it should be done in this PR. Therefore, I proposed options that will make easier the movement of this variable to the config file.

Reply:
I messed up with config